protocol/packages/protocol/src/lib.rs

```diff
@@ -0,0 +1,13 @@
+use std::collections::BTreeSet;
+
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub struct Protocol {
+    pub dex_network: String,
+    pub dex: String,
+    pub liquidity_provider_currency_ticker: String,
+    pub stable_currency_ticker: String,
+    pub lease_currencies_tickers: BTreeSet<String>,
```

Comment:
We already have an abstraction that on the wire is presented with its ticker. That is CurrencyDTO<G>. Please use it.

To save a cycle - G is unknown if the package does not depend on currencies. Either move the type to another package or make it generic over G.